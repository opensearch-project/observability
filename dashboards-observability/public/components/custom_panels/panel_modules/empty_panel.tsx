/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiSpacer, EuiText, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
import React, { useState } from 'react';
import { AddVisualizationPopover } from '../helpers/add_visualization_popover';

/*
 * EmptyPanelView - This Sub-component is shown to the user when a operational panel is empty
 *
 * Props taken in as params are:
 * addVizDisabled -> Boolean to enable/disable the add visualization button
 * getVizContextPanels -> Function to populate the add visualization popover
 */

interface EmptyPanelViewProps {
  addButton?: any;
  addVizDisabled: boolean;
  showFlyout: (isReplacement?: boolean, replaceVizId?: string) => void;
}

export const EmptyPanelView = ({
  addVizDisabled,
  showFlyout,
  addButton = <AddVisualizationPopover addVizDisabled={addVizDisabled} showFlyout={showFlyout} />,
}: EmptyPanelViewProps) => {
  return (
    <div>
      <EuiSpacer size="xxl" />
      <EuiText data-test-subj="addFirstVisualizationText" textAlign="center">
        <h2>Start by adding your first visualization</h2>
        <EuiSpacer size="m" />
        <EuiText color="subdued" size="m">
          Use PPL Queries to fetch &amp; filter observability data and create visualizations
        </EuiText>
      </EuiText>
      <EuiSpacer size="m" />
      <EuiFlexGroup justifyContent="center">
        <EuiFlexItem grow={false}>{addButton}</EuiFlexItem>
      </EuiFlexGroup>
      <EuiSpacer size="xxl" />
    </div>
  );
};
