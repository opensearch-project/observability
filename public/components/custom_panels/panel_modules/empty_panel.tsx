/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import { EuiSpacer, EuiText, EuiFlexGroup, EuiFlexItem, EuiButton } from '@elastic/eui';
import React from 'react';

/*
* EmptyPanelView
* This Sub-component is shown to the user when a operational panel is empty
* Props:
* addVizWindow -> This function shows the add visualization window in operational panels view
*/

type Props = {
  addVizWindow: () => void;
  addVizDisabled: boolean;
};

export const EmptyPanelView = ({ addVizWindow, addVizDisabled }: Props) => {
  return (
    <div>
      <EuiSpacer size="xxl" />
      <EuiText textAlign="center">
        <h2>Start by adding your first visualization</h2>
        <EuiSpacer size="m" />
        <EuiText color="subdued" size="m">
          Use PPL Queries to fetch and filter Observability Data to Create Visualizations
        </EuiText>
      </EuiText>
      <EuiSpacer size="m" />
      <EuiFlexGroup justifyContent="center">
        <EuiFlexItem grow={false}>
          <EuiButton fullWidth={false} disabled={addVizDisabled} fill onClick={addVizWindow}>
            Add Visualization
          </EuiButton>
        </EuiFlexItem>
      </EuiFlexGroup>
      <EuiSpacer size="xxl" />
    </div>
  );
};
