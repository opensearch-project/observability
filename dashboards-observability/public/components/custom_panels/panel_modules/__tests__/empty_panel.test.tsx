/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { EmptyPanelView } from '../empty_panel';

describe('Empty panel view component', () => {
  configure({ adapter: new Adapter() });

  it('renders empty panel view with disabled popover', () => {
    const getVizContextPanels = jest.fn();
    const wrapper = mount(
      <EmptyPanelView addVizDisabled={true} getVizContextPanels={getVizContextPanels} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('EuiButton').prop('disabled')).toBe(true);
  });

  it('renders empty panel view with enabled popover', () => {
    const getVizContextPanels = jest.fn();
    const wrapper = mount(
      <EmptyPanelView addVizDisabled={false} getVizContextPanels={getVizContextPanels} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('EuiButton').prop('disabled')).toBe(false);
  });

});
