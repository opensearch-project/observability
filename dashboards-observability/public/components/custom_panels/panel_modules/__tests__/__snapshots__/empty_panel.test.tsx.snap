// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty panel view component renders empty panel view with disabled popover 1`] = `
<EmptyPanelView
  addVizDisabled={true}
  showFlyout={[MockFunction]}
>
  <div>
    <EuiSpacer
      size="xxl"
    >
      <div
        className="euiSpacer euiSpacer--xxl"
      />
    </EuiSpacer>
    <EuiText
      data-test-subj="addFirstVisualizationText"
      textAlign="center"
    >
      <div
        className="euiText euiText--medium"
        data-test-subj="addFirstVisualizationText"
      >
        <EuiTextAlign
          textAlign="center"
        >
          <div
            className="euiTextAlign euiTextAlign--center"
          >
            <h2>
              Start by adding your first visualization
            </h2>
            <EuiSpacer
              size="m"
            >
              <div
                className="euiSpacer euiSpacer--m"
              />
            </EuiSpacer>
            <EuiText
              color="subdued"
              size="m"
            >
              <div
                className="euiText euiText--medium"
              >
                <EuiTextColor
                  color="subdued"
                  component="div"
                >
                  <div
                    className="euiTextColor euiTextColor--subdued"
                  >
                    Use PPL Queries to fetch & filter observability data and create visualizations
                  </div>
                </EuiTextColor>
              </div>
            </EuiText>
          </div>
        </EuiTextAlign>
      </div>
    </EuiText>
    <EuiSpacer
      size="m"
    >
      <div
        className="euiSpacer euiSpacer--m"
      />
    </EuiSpacer>
    <EuiFlexGroup
      justifyContent="center"
    >
      <div
        className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <EuiFlexItem
          grow={false}
        >
          <div
            className="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <AddVisualizationPopover
              addVizDisabled={true}
              showFlyout={[MockFunction]}
            >
              <EuiPopover
                anchorPosition="downLeft"
                button={
                  <EuiButton
                    data-test-subj="addVisualizationButton"
                    disabled={true}
                    iconSide="right"
                    iconType="arrowDown"
                    onClick={[Function]}
                  >
                    Add visualization
                  </EuiButton>
                }
                closePopover={[Function]}
                display="inlineBlock"
                hasArrow={true}
                id="addVisualizationContextMenu"
                isOpen={false}
                ownFocus={true}
                panelPaddingSize="none"
              >
                <div
                  className="euiPopover euiPopover--anchorDownLeft"
                  id="addVisualizationContextMenu"
                >
                  <div
                    className="euiPopover__anchor"
                  >
                    <EuiButton
                      data-test-subj="addVisualizationButton"
                      disabled={true}
                      iconSide="right"
                      iconType="arrowDown"
                      onClick={[Function]}
                    >
                      <EuiButtonDisplay
                        baseClassName="euiButton"
                        data-test-subj="addVisualizationButton"
                        disabled={true}
                        element="button"
                        iconSide="right"
                        iconType="arrowDown"
                        isDisabled={true}
                        onClick={[Function]}
                        type="button"
                      >
                        <button
                          className="euiButton euiButton--primary euiButton-isDisabled"
                          data-test-subj="addVisualizationButton"
                          disabled={true}
                          onClick={[Function]}
                          style={
                            Object {
                              "minWidth": undefined,
                            }
                          }
                          type="button"
                        >
                          <EuiButtonContent
                            className="euiButton__content"
                            iconSide="right"
                            iconType="arrowDown"
                            textProps={
                              Object {
                                "className": "euiButton__text",
                              }
                            }
                          >
                            <span
                              className="euiButtonContent euiButtonContent--iconRight euiButton__content"
                            >
                              <EuiIcon
                                className="euiButtonContent__icon"
                                color="inherit"
                                size="m"
                                type="arrowDown"
                              >
                                <EuiIconEmpty
                                  aria-hidden={true}
                                  className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                                  focusable="false"
                                  role="img"
                                  style={null}
                                >
                                  <svg
                                    aria-hidden={true}
                                    className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                                    focusable="false"
                                    height={16}
                                    role="img"
                                    style={null}
                                    viewBox="0 0 16 16"
                                    width={16}
                                    xmlns="http://www.w3.org/2000/svg"
                                  />
                                </EuiIconEmpty>
                              </EuiIcon>
                              <span
                                className="euiButton__text"
                              >
                                Add visualization
                              </span>
                            </span>
                          </EuiButtonContent>
                        </button>
                      </EuiButtonDisplay>
                    </EuiButton>
                  </div>
                </div>
              </EuiPopover>
            </AddVisualizationPopover>
          </div>
        </EuiFlexItem>
      </div>
    </EuiFlexGroup>
    <EuiSpacer
      size="xxl"
    >
      <div
        className="euiSpacer euiSpacer--xxl"
      />
    </EuiSpacer>
  </div>
</EmptyPanelView>
`;

exports[`Empty panel view component renders empty panel view with enabled popover 1`] = `
<EmptyPanelView
  addVizDisabled={false}
  showFlyout={[MockFunction]}
>
  <div>
    <EuiSpacer
      size="xxl"
    >
      <div
        className="euiSpacer euiSpacer--xxl"
      />
    </EuiSpacer>
    <EuiText
      data-test-subj="addFirstVisualizationText"
      textAlign="center"
    >
      <div
        className="euiText euiText--medium"
        data-test-subj="addFirstVisualizationText"
      >
        <EuiTextAlign
          textAlign="center"
        >
          <div
            className="euiTextAlign euiTextAlign--center"
          >
            <h2>
              Start by adding your first visualization
            </h2>
            <EuiSpacer
              size="m"
            >
              <div
                className="euiSpacer euiSpacer--m"
              />
            </EuiSpacer>
            <EuiText
              color="subdued"
              size="m"
            >
              <div
                className="euiText euiText--medium"
              >
                <EuiTextColor
                  color="subdued"
                  component="div"
                >
                  <div
                    className="euiTextColor euiTextColor--subdued"
                  >
                    Use PPL Queries to fetch & filter observability data and create visualizations
                  </div>
                </EuiTextColor>
              </div>
            </EuiText>
          </div>
        </EuiTextAlign>
      </div>
    </EuiText>
    <EuiSpacer
      size="m"
    >
      <div
        className="euiSpacer euiSpacer--m"
      />
    </EuiSpacer>
    <EuiFlexGroup
      justifyContent="center"
    >
      <div
        className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <EuiFlexItem
          grow={false}
        >
          <div
            className="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <AddVisualizationPopover
              addVizDisabled={false}
              showFlyout={[MockFunction]}
            >
              <EuiPopover
                anchorPosition="downLeft"
                button={
                  <EuiButton
                    data-test-subj="addVisualizationButton"
                    disabled={false}
                    iconSide="right"
                    iconType="arrowDown"
                    onClick={[Function]}
                  >
                    Add visualization
                  </EuiButton>
                }
                closePopover={[Function]}
                display="inlineBlock"
                hasArrow={true}
                id="addVisualizationContextMenu"
                isOpen={false}
                ownFocus={true}
                panelPaddingSize="none"
              >
                <div
                  className="euiPopover euiPopover--anchorDownLeft"
                  id="addVisualizationContextMenu"
                >
                  <div
                    className="euiPopover__anchor"
                  >
                    <EuiButton
                      data-test-subj="addVisualizationButton"
                      disabled={false}
                      iconSide="right"
                      iconType="arrowDown"
                      onClick={[Function]}
                    >
                      <EuiButtonDisplay
                        baseClassName="euiButton"
                        data-test-subj="addVisualizationButton"
                        disabled={false}
                        element="button"
                        iconSide="right"
                        iconType="arrowDown"
                        isDisabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <button
                          className="euiButton euiButton--primary"
                          data-test-subj="addVisualizationButton"
                          disabled={false}
                          onClick={[Function]}
                          style={
                            Object {
                              "minWidth": undefined,
                            }
                          }
                          type="button"
                        >
                          <EuiButtonContent
                            className="euiButton__content"
                            iconSide="right"
                            iconType="arrowDown"
                            textProps={
                              Object {
                                "className": "euiButton__text",
                              }
                            }
                          >
                            <span
                              className="euiButtonContent euiButtonContent--iconRight euiButton__content"
                            >
                              <EuiIcon
                                className="euiButtonContent__icon"
                                color="inherit"
                                size="m"
                                type="arrowDown"
                              >
                                <EuiIconArrowDown
                                  aria-hidden={true}
                                  className="euiIcon euiIcon--medium euiIcon--inherit euiButtonContent__icon"
                                  focusable="false"
                                  role="img"
                                  style={null}
                                >
                                  <svg
                                    aria-hidden={true}
                                    className="euiIcon euiIcon--medium euiIcon--inherit euiButtonContent__icon"
                                    focusable="false"
                                    height={16}
                                    role="img"
                                    style={null}
                                    viewBox="0 0 16 16"
                                    width={16}
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M13.069 5.157L8.384 9.768a.546.546 0 01-.768 0L2.93 5.158a.552.552 0 00-.771 0 .53.53 0 000 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 000-.76.552.552 0 00-.771 0z"
                                      fillRule="non-zero"
                                    />
                                  </svg>
                                </EuiIconArrowDown>
                              </EuiIcon>
                              <span
                                className="euiButton__text"
                              >
                                Add visualization
                              </span>
                            </span>
                          </EuiButtonContent>
                        </button>
                      </EuiButtonDisplay>
                    </EuiButton>
                  </div>
                </div>
              </EuiPopover>
            </AddVisualizationPopover>
          </div>
        </EuiFlexItem>
      </div>
    </EuiFlexGroup>
    <EuiSpacer
      size="xxl"
    >
      <div
        className="euiSpacer euiSpacer--xxl"
      />
    </EuiSpacer>
  </div>
</EmptyPanelView>
`;
